### PR TITLE
CoreModuleInteraction Generalization

### PR DESCRIPTION
**Pull request by felix2feng**
- Add module functions: batchIncrementTokenOwnerModule, batchDecrementTokenOwnerModule, batchTransferBalanceModule, transferModule, batchTransferModule

Reducing Bytecode size
- Moved issuance calculations to a library
- Shortened Core error strings to decrease bytecode size

Misc
- Use `onlyModule` modifier in CoreModuleInteraction

### CODE REVIEW

File: deployments/stages/1_libraries.ts [hunk 1]
```diff
@@ -68,6 +69,18 @@ export class LibrariesStage implements DeploymentStageInterface {
     return await ExchangeIssueLibraryContract.at(address, this._web3, TX_DEFAULTS);
   }
 
+  private async deployIssuanceLibrary(): Promise<IssuanceLibraryContract> {
```
Comment (kermankohli) [inline deployments/stages/1_libraries.ts]: You'll need to update the tests file to ensure this gets deployed

File: deployments/stages/4_authorization.ts [hunk 2]
```diff
@@ -62,19 +62,8 @@ export class AuthorizationStage implements DeploymentStageInterface {
...
-      'RebalancingTokenIssuanceModule',
-    ]);
-
-    await this.addAuthorizedAddressesToTransferProxy([
-      'Core',
-      'ExchangeIssueModule',
-      'RebalanceAuctionModule',
-      'RebalancingTokenIssuanceModule',
-    ]);
+    await this.addAuthorizedAddressesToVault(['Core']);
```
Comment (kermankohli) [inline deployments/stages/4_authorization.ts]: Tests need to be updated for this too

File: contracts/core/extensions/CoreModuleInteraction.sol [hunk 3]
```diff
@@ -36,6 +36,18 @@ contract CoreModuleInteraction is
     CoreState,
     ReentrancyGuard
 {
+    modifier onlyModule() {
```
Comment (bweick) [inline contracts/core/extensions/CoreModuleInteraction.sol]: We've kind of eschewed the use of modifiers why start now? Does this help with bytesize? As far as I remember modifiers get compiled into the function anyway so this increases bytesize right?

Comment (felix2feng) [reply contracts/core/extensions/CoreModuleInteraction.sol]: In this case, it became really un-DRY to have the same modifier code for 8 functions

**Pull request merged**